modules/parachains/src/mock.rs

```diff
@@ -40,7 +36,108 @@ type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRunt
 
 pub const PARAS_PALLET_NAME: &str = "Paras";
 pub const UNTRACKED_PARACHAIN_ID: u32 = 10;
-pub const MAXIMAL_PARACHAIN_HEAD_SIZE: u32 = 512;
+pub const MAXIMAL_PARACHAIN_HEAD_DATA_SIZE: u32 = 1 + 8 + 32;
```

Comment:
nit:

```suggestion
// use exact expected encoded size: `vec_len_size + header_number_size + state_root_hash_size`
pub const MAXIMAL_PARACHAIN_HEAD_DATA_SIZE: u32 = 1 + 8 + 32;
```

also maybe I'm misunderstanding the `1 + 8 + 32` here, but isn't vec.len() supposed to be an usize (8 bytes)?

Reply:
Len is encoded as compact int, so for small values it is 1 byte